app/design/adminhtml/default/default/template/login.phtml

```diff
@@ -35,26 +33,95 @@
+                        <?= $this->getChildHtml('form.additional.info') ?>
+                        <div class="clear"></div>
+                        <div class="form-buttons">
+                            <a class="left" href="<?= Mage::helper('adminhtml')->getUrl('adminhtml/index/forgotpassword', ['_nosecret' => true]) ?>"><?= Mage::helper('adminhtml')->__('Forgot your password?') ?></a>
+                            <input type="submit" class="form-button" value="<?= Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" title="<?= Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" />
+                        </div>
                     </div>
-                    <div class="input-box input-right">
-                        <label for="login"><?= Mage::helper('adminhtml')->__('Password:') ?></label><br />
-                        <input type="password" id="login" name="login[password]" value="" class="required-entry input-text" />
-                    </div>
-                    <?= $this->getChildHtml('form.additional.info') ?>
-                    <div class="clear"></div>
-                    <div class="form-buttons">
-                        <a class="left" href="<?= Mage::helper('adminhtml')->getUrl('adminhtml/index/forgotpassword', ['_nosecret' => true]) ?>"><?= Mage::helper('adminhtml')->__('Forgot your password?') ?></a>
-                        <input type="submit" class="form-button" value="<?= Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" title="<?= Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" />
+                    <div id="step2" class="no-display">
+                        <div class="input-box input-left">
+                            <label for="twofa_verification_code"><?= Mage::helper('adminhtml')->__('2FA verification code:') ?></label><br />
+                            <input type="text" id="twofa_verification_code" name="login[twofa_verification_code]" value="" class="required-entry input-text validate-number" autocomplete="one-time-code" disabled />
+                        </div>
+                        <div class="clear"></div>
+                        <div class="form-buttons">
+                            <a id="back-button" class="left" href="#"><?= Mage::helper('adminhtml')->__('Back') ?></a>
+                            <input type="submit" class="form-button" value="<?= Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" title="<?= Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" />
+                        </div>
                     </div>
                 </div>
                 <p class="legal">Powered by Maho</p>
             </form>
             <div class="bottom"></div>
-            <script type="text/javascript">
-                var loginForm = new varienForm('loginForm');
+            <script type="module">
```

Comment:
question, why type module?

Reply:
it seems to work also with type text/javascript too

Reply:
Just creates a new scope, that way nothing can overwrite the methods, and we have no possibility of something like `const validator` clashing with another variable defined elsewhere. You probably saw I used that a lot in the ExtJS PR.

Reply:
I realize I don't know anything anymore 😂

Reply:
All this stuff that you couldn't use a few years ago is now is widely supported, I had to catch up since I haven't done frontend devel in some time.

Basically my thought was to use module where possible, that way you can do `const form = new VarienForm()` and not worry about whatever other scripts are on the page.

Sometimes we need stuff in the global scope though, cause Magento likes to make it's form buttons use `onclick="myFunction()"`. So I either did `window.myFunction = function() {}` or just used a regular script tag. HTML5 also lets us remove the `type="text/javascript"` part too.

We should probably standardize this all, but for now there's just sooo many inline script tags....

Reply:
> Basically my thought was to use module where possible, that way you can do `const form = new VarienForm()` and not worry about whatever other scripts are on the page.

it's a nice idea for sure

> We should probably standardize this all, but for now there's just sooo many inline script tags....

yeah.....